src/transactions/transactionBuilder/transactionBuilder.ts

```diff
@@ -189,7 +202,7 @@ export async function generateViewFunctionPayload(args: InputViewFunctionDataWit
   });
 
   // Fill in the ABI
-  return generateViewFunctionPayloadWithABI({ abi: functionAbi, ...args });
+  return generateViewFunctionPayloadWithABI({ ...args, abi: functionAbi });
```

Comment:
yeah oops I actually meant to note this in the PR, I'm not sure if it's intended or not, but basically if you pass in your own `abi` it overwrites the fetched ABI because of the order of the spread on ...args

basically if you pass `abi: undefined` it ignores the fetched ABI and tries to use `undefined`

It's kind of a small thing, since most often people won't pass `abi: undefined`, but in my case I was doing that, I just figured if you're fetching it it should use the results